amplify/backend/function/getMembersForColony/src/index.js

```diff
@@ -121,177 +121,142 @@ exports.handler = async (event) => {
     );
   }
 
+  const allColonyWatchers = data?.getColonyByAddress?.items[0]?.watchers.items;
+
   // Incase there are members but no reputation on the colony,
   // we can stop here and return the list of watchers
-  if (!addressesWithReputation?.length) {
+  if (!contributorAddresses.size) {
+    if (domainId > Id.RootDomain) {
+      return {
+        contributors: [],
+        watchers: [], // You can only "watch" the root domain.
+      };
+    }
+
     return {
       contributors: [],
-      watchers:
-        domainId > Id.RootDomain
-          ? [] // There will be no Watchers outside of the root domain
-          : data?.getColonyByAddress?.items[0]?.watchers.items.map(
-              ({ user }) => ({
-                address: user.id,
-                user,
-              }),
-            ),
+      watchers: allColonyWatchers.map(({ user }) => ({
+        address: user.id,
+        user,
+      })),
     };
   }
 
-  // Identify watchers & contributors
-  data?.getColonyByAddress?.items[0]?.watchers.items.forEach((item) => {
-    if (
-      addressesWithReputation?.some(
-        (address) => address.toLowerCase() === item.user.id.toLowerCase(),
-      )
-    ) {
-      contributors.push({
-        address: item.user.id,
-        user: item.user,
-      });
-    } else {
-      watchers.push({
-        address: item.user.id,
-        user: item.user,
-      });
-    }
-  });
-
-  // There will be no Watchers outside of the root domain
-  if (domainId > Id.RootDomain) {
-    watchers.length = 0;
-  }
-
-  // now catch addresses that have reputation but are not in the watchers list
-  // i.e. address that was awarded reputation but has not joined the colony
-  if (addressesWithReputation?.length !== contributors?.length) {
```

Comment:
I presume this is responsible for my duplicate contributor, probably combined with some specific user flow like being awarded rep while not joined to a colony and then joining... or something like that. It's hard to tell. 

I have changed so that we generate contributors straight from the source of truth provided by the chain, instead of trying to splice together two separate lists. This should solve the problem.